tree/ntuple/v7/test/ntuple_descriptor.cxx

```diff
@@ -555,12 +555,18 @@ TEST(RNTupleDescriptor, BuildStreamerInfos)
       descBuilder.AddField(fieldBuilder.FieldId(1).MakeDescriptor().Unwrap());
       descBuilder.AddFieldLink(0, 1);
       int i = 2;
-      // In this test, we only support leaf fields or fields with only direct sub fields
-      for (const auto &subField : field.GetSubFields()) {
-         fieldBuilder = RFieldDescriptorBuilder::FromField(*subField);
+      // In this test, we only support field hierarchies up to 2 levels
+      for (const auto &child : field.GetSubFields()) {
+         fieldBuilder = RFieldDescriptorBuilder::FromField(*child);
          descBuilder.AddField(fieldBuilder.FieldId(i).MakeDescriptor().Unwrap());
          descBuilder.AddFieldLink(1, i);
          i++;
+         for (const auto &grandChild : child->GetSubFields()) {
+            fieldBuilder = RFieldDescriptorBuilder::FromField(*grandChild);
+            descBuilder.AddField(fieldBuilder.FieldId(i).MakeDescriptor().Unwrap());
+            descBuilder.AddFieldLink(1, i);
```

Comment:
Hm, in principle this should be the outer field, no? (probably doesn't make a difference because we never write nor read the ntuple)